test/docker-e2e/e2e_state_sync_test.go

```diff
@@ -214,3 +210,423 @@ func (s *CelestiaTestSuite) TestStateSyncMocha() {
+	}
+
+	return false, false, fmt.Errorf("could not fetch Prometheus metrics from any endpoint")
+}
+
+// analyzeSyncMethod determines if state sync or block sync was used based on height progression
+func analyzeSyncMethod(t *testing.T, heightHistory []int64) (usedStateSync bool, usedBlockSync bool) {
+	if len(heightHistory) < 2 {
+		t.Logf("Insufficient height data for analysis: %v", heightHistory)
+		return false, false
+	}
+
+	startHeight := heightHistory[0]
+	endHeight := heightHistory[len(heightHistory)-1]
+
+	// State sync pattern: sudden jump from 0 to high number (minimal intermediate values)
+	// Block sync pattern: incremental progression through many intermediate heights
+
+	if len(heightHistory) <= 3 && startHeight == 0 && endHeight > 50 {
+		// Very few data points with sudden jump = state sync
+		t.Logf("Detected state sync: sudden jump from %d to %d with %d data points",
+			startHeight, endHeight, len(heightHistory))
+
+		// Additional validation: ensure it's a true jump, not just fast incremental sync
+		if len(heightHistory) == 2 && heightHistory[1] > heightHistory[0]+20 {
+			t.Logf("Confirmed state sync: classic pattern [0, %d] with large jump", heightHistory[1])
+			return true, false
+		}
+
+		return true, false
+	}
+
+	if len(heightHistory) > 5 {
+		// Many data points = likely incremental block sync
+		intermediateCount := 0
+		for i := 1; i < len(heightHistory)-1; i++ {
+			if heightHistory[i] > 0 && heightHistory[i] < endHeight-10 {
+				intermediateCount++
+			}
+		}
+
+		if intermediateCount > 2 {
+			t.Logf("Detected block sync: incremental progression with %d intermediate heights", intermediateCount)
+			return false, true
+		}
+
+		t.Logf("Pattern suggests state sync: few intermediate heights despite %d data points", len(heightHistory))
+		return true, false
+	}
+
+	// moderate data points - analyze for incremental pattern
+	hasIncremental := false
+	for i := 1; i < len(heightHistory); i++ {
+		if heightHistory[i] > 0 && heightHistory[i-1] >= 0 {
+			diff := heightHistory[i] - heightHistory[i-1]
+			if diff > 0 && diff < 20 && heightHistory[i-1] > 0 {
+				hasIncremental = true
+				break
+			}
+		}
+	}
+
+	if hasIncremental {
+		t.Logf("Detected block sync: incremental progression pattern in %v", heightHistory)
+		return false, true
+	}
+
+	t.Logf("Detected state sync: no incremental pattern in %v", heightHistory)
+	return true, false
+}
+
+// verifySyncMethod validates that state sync was used and not block sync
+func verifySyncMethod(t *testing.T, heightHistory []int64, stateSyncNode *tastoradockertypes.ChainNode) {
+	t.Logf("Height progression during sync: %v", heightHistory)
+
+	// Analyze height progression pattern
+	usedStateSync, usedBlockSync := analyzeSyncMethod(t, heightHistory)
+
+	// FAIL if block sync was detected
+	if usedBlockSync {
+		t.Fatalf("CRITICAL FAILURE: Node used BLOCK SYNC instead of STATE SYNC! "+
+			"This test requires state sync. Height progression: %v", heightHistory)
```

Comment:
IMO the all caps aren't necessary

```suggestion
		t.Fatalf("Failed because state sync node used block sync instead of state sync. Height progression: %v", heightHistory)
```